### PR TITLE
Add step binary

### LINKED ISSUE #1746: Proposal: Add new binary cmd/step
SQLFlow needs the capability of translating and executing each SQL statement, with or without the extending syntax.  This capability makes the Argo/Tekton workflow tracable; it also makes it possible to integrate SQLFlow with IDEs like Visual Studio Code to support step-by-step running.

Currently, we rely on the binary `cmd/repl` as the step runner.  However, `cmd/repl` also has other roles like a local debug mode of SQLFlow and runs a SQL program instead of a SQL statement.

I agree that we keep `cmd/repl` as a SQLFlow server client, just like the Jupyter Notebook client.  But to make its role as a client clear, we need a new tool to run each step in the Argo/Tekton workflow.

### PR DESCRIPTION
**Pull request by Yancey0623**
fixed #1746 

TODOs:
1. Add more test cases.
1. using the `step` binary in the workflow.

### CODE REVIEW

File: cmd/step/main.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"regexp"
+	"time"
+
+	"sqlflow.org/sqlflow/cmd/repl/tablewriter"
```
Comment (typhoonzero) [inline cmd/step/main.go:25]: Move `tablewriter` under `pkg`? Seems it becomes a lib.

File: cmd/step/main.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+	defer func() {
+		if e := table.Flush(); e != nil {
+			log.Fatal(e)
+		}
+		fmt.Printf("(%.2f sec)\n", float64(time.Now().UnixNano()-startTime)/1e9)
+		fmt.Println()
+	}()
+
+	// discard the log output here just because using both log and pipe writer may mix the output
```
Comment (typhoonzero) [inline cmd/step/main.go:86]: Output everything using `log` may fix this issue.

**Review by typhoonzero — APPROVED** [review]

File: cmd/step/main.go (hunk 1 above)
Comment (Yancey0623) [reply cmd/step/main.go:25]: Good idea.

File: cmd/step/main.go (hunk 2 above)
Comment (Yancey0623) [reply cmd/step/main.go:86]: I see, but there is something need to polish:
1. `RunSQLProgram` return error
1. `tablewriter` can output using log

I will do that in the next PR.

**Pull request merged**